version: '3.7'
services:
  mongodb:
    image: docker.io/mongo:latest
    ports:
      - "27017:27017"
    hostname:
      mongodb
    volumes:
      - ./data/mongo:/data/db
    networks:
      - backend
  postgres:
    image: docker.io/postgres:15.2-alpine3.17
    ports:
      - "5432:5432"
    hostname:
        postgres
    environment:
        POSTGRES_USER: postgres
        POSTGRES_PASSWORD: postgres
        POSTGRES_DB: ipaas
    volumes:
      - ./data/pg:/var/lib/postgresql/data
    networks:
      - backend
  redis:
    image: docker.io/redis:7.0-alpine3.17
    ports:
      - "6379:6379"
    volumes:
      - ./data/redis:/data
    networks:
      - backend
  app:
    build: .
    volumes:
      - ./:/home/node/app
    ports:
      - "3000:3000"
    restart: on-failure
    depends_on:
      - mongodb
      - redis
      - postgres
    networks:
      - backend

networks:
  backend:
    driver: bridge